### PR TITLE
Req 59 oauth setup

### PR DESCRIPTION
**Pull request by holingpoon**
Namespace is now centralized to `NYPL\HoldRequestResultConsumer` because it is what makes most sense at the moment. 

All code associated with how a `Patron` is fetched from NYPL Platforms API is currently written to prove that the `OAuthClient` worked, logging into the API is successful, returning a token, and the token can be used to start getting data.

### CODE REVIEW

File: src/OAuthClient/APIClient.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public static function patch($uri = '', array $options = [])
+    {
+        return self::getClient()->patch(
+            $uri,
+            self::getOptions($options)
+        );
+    }
+}
```
Comment (gkallenberg) [inline src/OAuthClient/APIClient.php:111]: Are get, put, post, patch operations necessary here?

Comment (gkallenberg) [reply src/OAuthClient/APIClient.php:111]: Changes seem good to me. Are you publishing to an email service or actually sending emails?

Comment (holingpoon) [reply src/OAuthClient/APIClient.php:111]: Hi @gkallenberg! So for Bib and Patron info I would only need a `get`, but updating `HoldRequestService` I'll need a `post` to update the data. Would `post`s be available on that layer?

For e-mails, I'm just sending e-mails directly to users now from either the one provided in `DocDeliveryData` or via a `Patron` lookup.

File: src/Config/Config.sample.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+namespace NYPL\HoldRequestResultConsumer\Config;
+
+class Config
+{
+    const API_BASE_URL = 'https://api.nypltech.org/api/v0.1';
```
Comment (rhernand3z) [inline src/Config/Config.sample.php]: Wondering if these URL's should be within the repo or as ENV vars.

**Review by rhernand3z — APPROVED** [review]
Overall looks good.

File: src/Config/Config.sample.php (hunk 2 above)
Comment (holingpoon) [reply src/Config/Config.sample.php]: Good call! Yeah, I think for security and privacy, it'll make more sense to make them environment variables. Thanks!

**Pull request merged**